File: ITapply/ITapply.Services/Database/Application.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using static ITapply.Services.Database.Enums;

namespace ITapply.Services.Database
{
    public class Application
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int CandidateId { get; set; }
        [ForeignKey("CandidateId")]
        public Candidate Candidate { get; set; }

        [Required]
        public int JobPostingId { get; set; }
        [ForeignKey("JobPostingId")]
        public JobPosting JobPosting { get; set; }

        [Required]
        public DateTime ApplicationDate { get; set; } = DateTime.Now;

        [Required]
        public ApplicationStatus Status { get; set; } = ApplicationStatus.Applied; // Default status

        [StringLength(5000)]
        public string CoverLetter { get; set; }

        [Required]
        public int CVDocumentId { get; set; }
        [ForeignKey("CVDocumentId")]
        public CVDocument CVDocument { get; set; } // Reference to the specific CV document used for this application

        [Required]
        [StringLength(100)]
        public string Availability { get; set; }

        [StringLength(2000)]
        public string InternalNotes { get; set; } // For Employer's internal use

        [StringLength(2000)]
        public string EmployerMessage { get; set; } // Message from Employer to Candidate

        public bool ReceiveNotifications { get; set; } = true;
    }
}

File: ITapply/ITapply.UI/itapply_desktop/lib/models/application.dart
import 'package:itapply_desktop/models/enums.dart';
import 'package:json_annotation/json_annotation.dart';

part 'application.g.dart';
@JsonSerializable(explicitToJson: true)
class Application {
  final int id;
  final int candidateId;
  final String? candidateName;
  final String? candidateEmail;
  final int jobPostingId;
  final String? jobTitle;
  final String? companyName;
  final DateTime applicationDate;
  @JsonKey(unknownEnumValue: ApplicationStatus.pending)
  final ApplicationStatus status;
  final String? coverLetter;
  final int cvDocumentId;
  final String? cvDocumentName;
  final String? availability;
  final String? internalNotes;
  final String? employerMessage;
  final bool receiveNotifications;
  
  Application({
    required this.id,
    required this.candidateId,
    this.candidateName,
    this.candidateEmail,
    required this.jobPostingId,
    this.jobTitle,
    this.companyName,
    required this.applicationDate,
    required this.status,
    this.coverLetter,
    required this.cvDocumentId,
    this.cvDocumentName,
    this.availability,
    this.internalNotes,
    this.employerMessage,
    required this.receiveNotifications,
  });

  factory Application.fromJson(Map<String, dynamic> json) => _$ApplicationFromJson(json);
  Map<String, dynamic> toJson() => _$ApplicationToJson(this);
}